wagtail/wagtailadmin/static_src/wagtailadmin/js/page-editor.js

```diff
@@ -391,6 +391,28 @@ $(function() {
         insertRichTextDeleteControl(this);
     });
 
+    /* Make user confirm before leaving the editor if there are unsaved changes */
+    var initialData = $('#page-edit-form').serialize();
+    function dataHasChanged() {
+        return $('#page-edit-form').serialize() != initialData;
+    }
+
+    window.addEventListener('beforeunload', function(event) {
+        // Ignore if the user clicked one of the submit buttons
+        var trigger = $(event.target.activeElement);
+        var triggeredBySubmitButton = false;
+        $('#page-edit-form button[type="submit"]').each(function() {
+            if ($(this).is(trigger)) {
+                triggeredBySubmitButton = true;
+            }
+        });
+
+        if (!triggeredBySubmitButton && dataHasChanged()) {
+            event.returnValue = '\o/';
+            return '\o/';
```

Comment:
Empty string works in Chrome like shown below, but doesn't work in Firefox.

<img width="532" alt="screen shot 2016-01-28 at 13 37 53" src="https://cloud.githubusercontent.com/assets/509198/12642063/8713275e-c5c4-11e5-821b-80479f62ae6c.png">